examples/benchmark.rs

```diff
@@ -1,3 +1,5 @@
+#![allow(clippy::collapsible_if)]
```

Comment:
Should we just disable clippy in examples?

Reply:
I tend to look at the examples first when I'm looking at a new project, so my _personal_ preference would be to make sure that the examples are just as tidy. But I'm happy to drop this if you'd rather not bother.

Reply:
No objection if you prefer it this way.